doc/rm/c_cpp_coding_style.md

```diff
@@ -162,6 +162,18 @@ Any nonstandard features that are used must be compatible with both GCC and Clan
 
 This rule deviates from the Google C++ style guide to align closer with a typical way of writing C code.
 
+The names of enumeration constants must be prefixed with the name of their respective enumeration type.
```

Comment:
Do we really want a rule this strict? I'm in favor of something mildly wimpier, since I've been writing
```
typedef enum dif_<dev>_<fn>_result {
  kDif<Dev><Fn>Ok = 0,
  kDif<Dev><Fn>Bad,
} dif_<dev>_<fn>_result_t;
```

I might argue that what we *actually* want is something like "all symbols in a particular header must share the same, unique prefix" (with fundamental libraries being excempt, of course).

Reply:
"All symbols in a particular header must share the same, unique prefix" is certainly worth adding for the same namespacing issues.

I'm not sure we want to relax the enum naming rule (given there have been DIF review comments from @gkelly to move from your example style to the rule's style). I would like input from others on how strict they want this rule before we commit it.

Reply:
My vote is for the strict one, i.e. `kMyWonderfulOption{Enabled,Disabled,OnlySometimes}` for `my_wonderful_option_t`.